p2p/src/test/kotlin/maru/p2p/P2PTest.kt

```diff
@@ -148,8 +148,9 @@ class P2PTest {
       p2pNetwork1
         .getPeer(LibP2PNodeId(PeerId.fromBase58(PEER_ID_NODE_2)))
         .get()
-        .disconnectCleanly(DisconnectReason.TOO_MANY_PEERS)
-        .thenPeek { assertNetworkHasPeers(p2pNetwork1, 0) }
+        .disconnectImmediately(Optional.empty(), true)
```

Comment:
Why is this better than using the disconnect cleanly? Thought the clean disconnect would be better

Reply:
The problem with the clean one is that it might take longer, and the time for the await starts ticking right after the call to close the connection. That is why I use the immediate now and await the 0 connections after.

Reply:
That is even better!! Done!